packages/ra-core/src/auth/useLogoutIfAccessDenied.ts

```diff
@@ -102,14 +111,8 @@ const useLogoutIfAccessDenied = (): LogoutIfAccessDenied => {
 
                     if (logoutUser) {
                         logout({}, redirectTo);
-                    } else {
-                        if (redirectTo.startsWith('http')) {
-                            // absolute link (e.g. https://my.oidc.server/login)
-                            window.location.href = redirectTo;
-                        } else {
-                            // internal location
-                            navigate(redirectTo);
-                        }
+                    } else if (redirectTo) {
+                        handleRedirect(redirectTo);
```

Comment:
Also, I realize the dependency array of the `useCallback` is no longer up-to-date: you need to remove `navigate` and instead add `handleRedirect`